src/MessagePack.Experimental/UnsafeBlittableFormatter/DynamicUnsafeUnmanagedStructResolver.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) All contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using MessagePack.Formatters;
+
+namespace MessagePack.Resolvers
+{
+    public sealed class DynamicUnsafeUnmanagedStructResolver : IFormatterResolver
+    {
+        public static readonly DynamicUnsafeUnmanagedStructResolver Instance = new DynamicUnsafeUnmanagedStructResolver();
+
+        private static readonly MethodInfo BaseIsReferenceOrContainsReferences = typeof(RuntimeHelpers).GetMethod(nameof(RuntimeHelpers.IsReferenceOrContainsReferences)) ?? throw new NullReferenceException();
+        private static readonly Type BaseFormatterCache = typeof(FormatterCache<>);
+        private static readonly Type BaseStruct = typeof(UnsafeUnmanagedStructFormatter<>);
+        private static readonly Type BaseArray = typeof(UnsafeUnmanagedStructArrayFormatter<>);
+
+        [ThreadStatic] private static Type[]? one;
+
+        static DynamicUnsafeUnmanagedStructResolver()
+        {
+            var currentDomain = AppDomain.CurrentDomain;
+            var assemblies = currentDomain.GetAssemblies();
+            foreach (var assembly in assemblies)
+            {
+                ProcessAssembly(assembly);
+            }
+
+            currentDomain.AssemblyLoad += CurrentDomainOnAssemblyLoad;
+        }
```

Comment:
AssemblyLoad handlers have such broad impact, I avoid them if at all possible. Why do we have to implement it this way as opposed to scanning an assembly that defines some type that we're serializing?

Reply:
Thinking more on this, we already have `MessagePackFormatterAttribute` which can be used to specify a formatter for a particular type or member. Can't that be used instead of a new module-level attribute?

Reply:
I agree with you.